ubuntu22-py38/Dockerfile

```diff
@@ -0,0 +1,36 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+# syntax=docker/dockerfile:1
+FROM ubuntu:22.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends curl gcc git rsync vim software-properties-common gpg-agent \
+  && apt-add-repository -y ppa:deadsnakes/ppa \
+  && apt update \
+  && apt install -y python3.8 python3.8-distutils python3-pip \
+  && update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 \
+  && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1 \
```

Comment:
maybe in a future iteration, we can make away with these commands and simply use the default Python that comes out of the box with Ubuntu 22.04... as we are using it as a base image. Alternatively we can provide dev container images using `python-3.8` as base directly